projects/plugins/jetpack/extensions/blocks/ai-paragraph/edit.js

```diff
@@ -4,222 +4,263 @@ import apiFetch from '@wordpress/api-fetch';
-			const content = res.prompts[ 0 ].text;
+function ShowLittleByLittle( { html, showAnimation } ) {
+	const [ content, setContent ] = useState( '' );
+	const [ contentSet, setContentSet ] = useState( false );
+
+	if ( ! contentSet ) {
+		// That will only be used once
+		if ( showAnimation ) {
 			// This is to animate text input. I think this will give an idea of a "better" AI.
 			// At this point this is an established pattern.
-			const tokens = content.split( ' ' );
-			for ( let i = 0; i < tokens.length; i++ ) {
+			const tokens = html.split( ' ' );
+			for ( let i = 1; i < tokens.length; i++ ) {
 				const output = tokens.slice( 0, i ).join( ' ' );
-				setTimeout( () => setAttributes( { content: output } ), 50 * i );
+				setTimeout( () => setContent( output ), 50 * i );
 			}
-			setTimeout( () => setAttributes( { content: content } ), 50 * tokens.length );
-		} )
-		.catch( () => {
-			setErrorMessage(
-				__(
-					'Whoops, we have encountered an error. AI is like really, really hard and this is an experimental feature. Please try again later.',
-					'jetpack'
-				)
-			);
-			setLoadingCompletion( false );
-		} );
-}
-
-function createPrompt( title = '', content = '', categoryNames = '' ) {
-	// If title is not added, we will only complete.
-	if ( ! title ) {
-		return content;
+			setTimeout( () => setContent( html ), 50 * tokens.length );
+		} else {
+			setContent( html );
+		}
+		setContentSet( true );
 	}
 
-	// Title, content and categories
-	if ( content && categoryNames.length ) {
-		return sprintf(
-			/** translators: This will be a prompt to OpenAI to generate a post based on the post title, comma-seperated category names and the last 240 characters of content. */
-			__( "This is a post titled '%1$s', published in categories '%2$s':\n\n … %3$s", 'jetpack' ), // eslint-disable-line @wordpress/i18n-no-collapsible-whitespace
-			title,
-			categoryNames,
-			content
-		);
-	}
+	return (
+		<>
+			<div className="content">
+				<RawHTML>{ content }</RawHTML>
+			</div>
+		</>
+	);
+}
 
-	// No content, only title and categories
-	if ( categoryNames.length ) {
-		return sprintf(
-			/** translators: This will be a prompt to OpenAI to generate a post based on the post title, and comma-seperated category names. */
-			__( "This is a post titled '%1$s', published in categories '%2$s':\n", 'jetpack' ), // eslint-disable-line @wordpress/i18n-no-collapsible-whitespace
-			title,
-			categoryNames
-		);
-	}
+export default function Edit( { attributes, setAttributes, clientId } ) {
+	const [ loadingCompletion, setLoadingCompletion ] = useState( false );
+	const [ loadingCategories, setLoadingCategories ] = useState( false );
+	const [ showAnimation, setShowAnimation ] = useState( false );
+	const [ needsMoreCharacters, setNeedsMoreCharacters ] = useState( 0 );
+	const [ showRetry, setShowRetry ] = useState( false );
+	const [ completionFinished, setCompletionFinished ] = useState( !! attributes.content );
```

Comment:
Can't you just watch `attributes.content` instead of adding a tracking state? _This might be my react noviceness showing._

Reply:
I need it for when the block `attributes.content` has already been filled previously, and the page is re-loaded for example. Otherwide it would display the blocks word-by-word at each page (re)load

Reply:
You are right. It is also a relic of [the previously discussed state](https://github.com/Automattic/jetpack/pull/28483#discussion_r1083899489)